### ISSUE TITLE
URL setters test

### ISSUE THREAD

**Issue by rubys** [w3c/web-platform-tests#1408]
tl;dr: http://intertwingly.net/projects/pegurl/url-setters.html

Longer version: I don't notice any URL setters tests in http://w3c.github.io/test-results/url/all.html

I've been developing some as a part of my reference implementation work: https://github.com/rubys/url/blob/peg.js/reference-implementation/test/urlsettest.yml

I believe that some form of this belongs in web-platform-tests.

I'm interested in any and all feedback, including but not limited to:
- what changes should be made to fit in with web-platform-tests
- identification of tests that don't match the spec
- changes needed to the spec to match deployed reality
- suggestions for additional tests.  Note the TODOs at the top of the YAML file for items I already plan to get to.

@annevk @sideshowbarker @jgraham

**Comment by annevk** [comment]
This got fixed by #2830.
